experiments/calibration/observation_map.jl

```diff
@@ -1,77 +1,215 @@
-- `variable_list`: A vector of variable names (strings) to extract and process.
-- `spinup_period`: A `Period` (default: `Year(1)`) to offset the analysis window from the simulation start date.
-
-# Returns
-- A flat vector containing seasonal mean values for each variable at each training location, concatenated
-  across all variables and locations.
-
-# Notes
-- Assumes that the simulation starts in December to allow clean extraction of full seasons starting with DJF.
-- The function uses a 1-year window starting from `start_date + spinup_period` to extract data.
-- Observational data are computed as seasonal means and flattened into a single vector.
-
-# Throws
-- An `AssertionError` if the simulation start date is not in December.
-
-# Example
-```julia
-simdir = load_simulation("path/to/output")
-locations = [(0.0, 45.0), (10.0, 50.0)]
-vars = ["lhf", "shf"]
-output = process_member_data(simdir, locations, vars)
+    process_member_data(diagnostics_folder_path, short_names, current_minibatch)
+
+Process the data of a single ensemble member and return a single column of the
+G ensemble matrix.
 """
 function process_member_data(
-    simdir,
-    training_locations,
-    variable_list;
-    spinup_period = Year(1),
+    diagnostics_folder_path,
+    short_names,
+    current_minibatch,
 )
+    sample_date_ranges = CALIBRATE_CONFIG.sample_date_ranges
+    nelements = CALIBRATE_CONFIG.nelements
+    @info "Short names: $short_names"
+    era5_obs_vars = LandSimVis.get_era5_obs_var_dict()
+    for short_name in short_names
+        short_name in keys(era5_obs_vars) || error(
+            "Variable $short_name does not appear in the observation dataset. Add the variable to get_era5_obs_var_dict",
+        )
+    end
 
-    # For later, for more flexible simulation times,
-    # vardates = ClimaAnalysis.Utils.time_to_date.(Ref(start_date), ClimaAnalysis.times(lhf))
-    # vardates[findlast(x->month(x)==11, vardates - Month(1))-1]
-
-    start_date =
-        DateTime(get(simdir, variable_list[1]).attributes["start_date"])
-    left = start_date + spinup_period
-    right = start_date + spinup_period + Year(1)
-    data = Dict(
-        var => window(
-            shift_to_start_of_previous_month(get(simdir, var)),
-            "time";
-            left,
-            right,
-        ) for var in variable_list
-    )
+    sim_var_dict = LandSimVis.get_sim_var_dict(diagnostics_folder_path)
+    vars = map(short_names) do short_name
+        var = sim_var_dict[short_name]()
+        var = ClimaAnalysis.average_season_across_time(var, ignore_nan = false)
+
+        ocean_mask = make_ocean_mask(nelements)
+        var = ocean_mask(var)
+
+        # Replace all NaNs on land with the mean value
+        # This is needed to stop small NaN values from stopping the calibration
+        # TODO: Replace this with mask aware flatten once it is merged in
+        # ClimaAnalysis
+        nanmean_land_val = 500.0
```

Comment:
Is this an appropriate fill value? It looks like this will apply the same value to different vars with potentially different scales. Maybe we could have a dict/mapping from short names to default fill values, with one overall default if the mapping is not found.